app/src/lib/stores/helpers/tutorial-assessor.ts

```diff
@@ -0,0 +1,145 @@
+  private createPRSkipped: boolean = getBoolean(skipCreatePullRequestKey, false)
+
+  public constructor(
+    /** Method to call when we need to re-check for an editor */
+    private resolveCurrentEditor: () => Promise<void>,
+    /** Method to call when we need to get the current editor */
+    private getResolvedExternalEditor: () => ExternalEditor | null
+  ) {}
+
+  /** Determines what step the user needs to complete next in the Onboarding Tutorial */
+  public async getCurrentStep(
+    isTutorialRepo: boolean,
+    repositoryState: IRepositoryState
+  ): Promise<TutorialStep> {
+    if (!isTutorialRepo) {
+      return TutorialStep.NotApplicable
+    } else if (!(await this.isEditorInstalled())) {
+      return TutorialStep.PickEditor
+    } else if (!this.isBranchCheckedOut(repositoryState)) {
+      return TutorialStep.CreateBranch
+    } else if (!this.hasChangedFile(repositoryState)) {
+      return TutorialStep.EditFile
+    } else if (!this.hasMultipleCommits(repositoryState)) {
+      return TutorialStep.MakeCommit
+    } else if (!this.commitPushed(repositoryState)) {
+      return TutorialStep.PushBranch
+    } else if (!this.pullRequestCreated(repositoryState)) {
+      return TutorialStep.OpenPullRequest
+    } else {
+      return TutorialStep.AllDone
+    }
+  }
+
+  private async isEditorInstalled(): Promise<boolean> {
+    if (this.installEditorSkipped || this.getResolvedExternalEditor()) {
+      return true
+    } else {
+      await this.resolveCurrentEditor()
+      return !!this.getResolvedExternalEditor()
+    }
+  }
+
+  private isBranchCheckedOut(repositoryState: IRepositoryState): boolean {
+    const { branchesState } = repositoryState
+    const { tip } = branchesState
+
+    const currentBranchName =
+      tip.kind === TipState.Valid ? tip.branch.name : null
+    const defaultBranchName =
+      branchesState.defaultBranch !== null
+        ? branchesState.defaultBranch.name
+        : null
+
+    return (
+      currentBranchName !== null &&
+      defaultBranchName !== null &&
+      currentBranchName !== defaultBranchName
+    )
+  }
+
+  private hasChangedFile(repositoryState: IRepositoryState): boolean {
+    if (this.hasMultipleCommits(repositoryState)) {
+      // User has already committed a change
+      return true
+    }
+    const { changesState } = repositoryState
+    return changesState.workingDirectory.files.length > 0
+  }
+
+  private hasMultipleCommits(repositoryState: IRepositoryState): boolean {
+    const { branchesState } = repositoryState
+    const { tip } = branchesState
+
+    // TODO: Verify with @niik that there will only be one commit initially
+    if (tip.kind === TipState.Valid) {
+      // For some reason sometimes the initial commit has a parent sha
+      // listed as an empty string...
+      // For now I'm filtering those out. Would be better to prevent that from happening
+      return tip.branch.tip.parentSHAs.filter(Boolean).length > 0
```

Comment:
I think passing `Boolean` here is a nifty trick but I doubt it's immediately obvious to a reader unless they're familiar with the intricacies of javascript's truthiness. Could we perhaps change this to `.some` instead?

```suggestion
      return tip.branch.tip.parentSHAs.some(x => x.length > 0)
```